model/infinispan/src/main/java/org/keycloak/cluster/infinispan/InfinispanNotificationsManager.java

```diff
@@ -68,58 +50,31 @@ public class InfinispanNotificationsManager {
 
     protected static final Logger logger = Logger.getLogger(InfinispanNotificationsManager.class);
 
-    private static final int BACKOFF_BASE_MILLIS = 10;
-    private static final int MAX_BACKOFF_RETRIES = 10;
-
     private final ConcurrentMultivaluedHashMap<String, ClusterListener> listeners = new ConcurrentMultivaluedHashMap<>();
 
     private final ConcurrentMap<String, TaskCallback> taskCallbacks = new ConcurrentHashMap<>();
 
     private final Cache<String, Object> workCache;
 
-    private final RemoteCache<String, Object> workRemoteCache;
-
     private final String myAddress;
 
     private final String mySite;
 
-    private final ExecutorService listenersExecutor;
-
-    protected InfinispanNotificationsManager(Cache<String, Object> workCache, RemoteCache<String, Object> workRemoteCache, String myAddress, String mySite, ExecutorService listenersExecutor) {
+    protected InfinispanNotificationsManager(Cache<String, Object> workCache, String myAddress, String mySite) {
```

Comment:
This class is now only used by the `InfinispanClusterProvider` and ~ 40% of lines are boiler plate (imports, variable declarations etc). Should we add the logic to `InfinispanClusterProvider` directly and remove this additional class?

Reply:
I would not do this change. The reason why this is done this way now is that `InfinispanNotificationManager` is initialzied only once on Keycloak startup in `lazyInit` method and then passed to each instance  of `InfinispanClusterProvider` that is reinitialized with each request.

The question is whether we really need to reinitialize it with each request and I would say we don't as the class is never using `KeycloakSession`. Still I would prefer to do this in a follow up. WDYT?

Reply:
Thanks for explaining. AFAICT you're right, a `InfinispanClusterProvider` shouldn't need to be created per request. This PR is already pretty big, so +1 to making the change in a follow-up.

Reply:
Follow up issue to track?

Reply:
Thanks @ryanemerson !